### DISCUSSION TITLE
How to use component in different framework

### DISCUSSION THREAD

**Discussion by tavisca-nilesh** [YogliB/svelte-component-template#1473] [General]
I want to use components built in svelte in microfrontend application. i have following questions
How can we use these components in angular?
how to generate bundle or npm package so that I can use it in any framework? npm run publish not working
could you please add some descriptions and roles of each file in the template?

**Comment by YogliB** [comment]
First off, when using Svelte components outside a Svelte app you'll need to [compile them](https://svelte.dev/docs#svelte_compile), so this template isn't the best choice and I think you should check out the [official template](https://github.com/sveltejs/component-template) which uses Rollup for the job.

after compiling and publishing, you can consume your components like so:
```javascript
import SvelteComponent from 'my-svelte-component';

// Angular specific
@ViewChild('myDiv') myDiv: ElementRef<HTMLDivElement>; 

// This will work only after AfterViewInit was called (or inside it)
const myComponent = new SvelteComponent({
    target: myDiv,
    props: {
    // assuming SvelteComponent.svelte contains something like
    // `export let answer`:
        answer: 42
    }
})
```

you can see the full API [here](https://svelte.dev/docs#Creating_a_component).

For publishing a component, [have a look here](https://www.youtube.com/watch?v=uapg3tstHSQ).

Hope this helps!

**Reply by tavisca-nilesh** [reply]
Got it. Thanks a lot.
